src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/BuildUpToDateCheck.cs

```diff
@@ -168,11 +182,11 @@ private void OnOutputGroupChanged(IImmutableDictionary<string, IOutputGroup> e)
             }
         }
 
-        private void OnChanged(IProjectVersionedValue<Tuple<IProjectSubscriptionUpdate, IProjectImportTreeSnapshot, IProjectSubscriptionUpdate, IImmutableDictionary<string, IOutputGroup>>> e)
+        private void OnChanged(IProjectVersionedValue<Tuple<IProjectSubscriptionUpdate, IProjectImportTreeSnapshot, IProjectSubscriptionUpdate, IImmutableDictionary<string, IOutputGroup>, IProjectItemSchema>> e)
         {
             OnProjectChanged(e.Value.Item1);
             OnProjectImportsChanged(e.Value.Item2);
-            OnSourceItemChanged(e.Value.Item3);
+            OnSourceItemChanged(e.Value.Item3, e.Value.Item5);
```

Comment:
Sure, will do. Item5 is the project schema which we use to interrogate about what item types are used for up to date input.